src/firefox/preferences.js

```diff
@@ -117,3 +119,38 @@ export function getPrefs(
     ...appPrefs,
   };
 }
+
+export function coerceCLICustomPreference(
+  cliPrefs: string | Array<string>
+): FirefoxPreferences {
+  let customPrefs = {};
+
+  if (!Array.isArray(cliPrefs)) {
+    cliPrefs = [cliPrefs];
+  }
+
+  for (let pref of cliPrefs) {
+    let [key, value] = pref.split('=');
+
+    if (/[^\w{@}.-]/.test(key)) {
+      throw new UsageError(`Invalid custom preference name: ${key}`);
+    }
+
+    if (value === `${parseInt(value)}`) {
+      value = parseInt(value, 10);
+    } else if (value === 'true' || value === 'false') {
+      value = (value === 'true');
+    }
+
+    let defaultProps = ['devtools.debugger.remote-enabled',
+      'devtools.debugger.prompt-connection', 'xpinstall.signatures.required'];
+    if (defaultProps.includes(key)) {
+      log.info(`Setting '${key}' is important for the work of WebExtensions` +
```

Comment:
`log.warn` is probably better here.
I would also use a shorter message, something like `'${key}' preference is used internally and it cannot be customized.`.